core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassConditionBuildItem.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.deployment.builditem.nativeimage;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Used to define a condition to register a class for reflection in native mode only when a specific type is reachable
+ */
+public final class ReflectiveClassConditionBuildItem extends MultiBuildItem {
+
+    private final String className;
+    private final String typeReachable;
```

Comment:
What exactly does this field represent? Is it a type that needs to be reachable in order to register the class for reflection? 

Maybe we could add an example to the javadoc?

Reply:
> What exactly does this field represent? Is it a type that needs to be reachable in order to register the class for reflection?

In this case, it is exactly what you understood. In practice, it could be used as a condition for any kind of metadata entry but for this use case, it is good enough (more details [here](https://www.graalvm.org/22.2/reference-manual/native-image/metadata/#specifying-metadata-with-json)).

Reply:
Done

Reply:
@mkouba correct. This is following the graalvm json schema (see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#specifying-metadata-with-json), not sure it's worth documenting in Quarkus as well.

Reply:
I think that we should at least add the link...